components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/service/AppGatewayAdmin.java

```diff
@@ -0,0 +1,442 @@
+        restClient.deleteNonVersionedWebAppForTenant(tenantDomain);
+    }
+
+    /**
+     * Delete non versioned webapp configuration from the gateway for super tenant users.
+     * @param appProviderName String.
+     * @param appName String.
+     * @param version String.
+     * @throws AppManagementException on errors.
+     */
+    public void deleteNonVersionedWebApp(String appProviderName, String appName, String version)
+            throws AppManagementException {
+        RESTAPIAdminClient restClient = new RESTAPIAdminClient(appProviderName, appName, version);
+        restClient.deleteNonVersionedWebApp();
+    }
+
+    /**
+     * Get non versioned webapp configuration from the gateway for tenant users.
+     * @param appProviderName String.
+     * @param appName String.
+     * @param version String.
+     * @param tenantDomain String.
+     * @return non versioned webapp configuration data.
+     * @throws AppManagementException on errors.
+     */
+    public AppData getNonVersionedWebAppDataForTenant(String appProviderName, String appName, String version,
+                                                      String tenantDomain) throws AppManagementException {
+        RESTAPIAdminClient restClient = new RESTAPIAdminClient(appProviderName, appName, version);
+        APIData apiData = restClient.getNonVersionedWebAppDataForTenant(tenantDomain);
+        return convert(apiData);
+    }
+
+    /**
+     * Get non versioned webapp configuration from the gateway for super tenant users.
+     * @param appProviderName String.
+     * @param appName String.
+     * @param version String.
+     * @return non versioned webapp configuration data.
+     * @throws AppManagementException on errors.
+     */
+    public AppData getNonVersionedWebAppData(String appProviderName, String appName, String version)
+            throws AppManagementException {
+        RESTAPIAdminClient restClient = new RESTAPIAdminClient(appProviderName, appName, version);
+        APIData apiData = restClient.getNonVersionedWebAppData();
+        return convert(apiData);
+    }
+
+    private AppData convert(APIData data) {
+        if (data == null) {
+            return null;
+        }
+        AppData apiData = new AppData();
+        apiData.setContext(data.getContext());
+        apiData.setFileName(data.getFileName());
+        apiData.setHost(data.getHost());
+        apiData.setName(data.getName());
+        apiData.setPort(data.getPort());
+        org.wso2.carbon.rest.api.stub.types.carbon.ResourceData[] resources = data.getResources();
+        List<ResourceData> resList = new ArrayList<ResourceData>();
+        if (resources != null && resources.length > 0) {
+            for (org.wso2.carbon.rest.api.stub.types.carbon.ResourceData res : resources) {
+                if (res == null) {
+                    continue;
+                }
+                ResourceData resource = convert(res);
+                resList.add(resource);
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/05d5cef1f9e3d525eb6d0c0af34943c96daac8ae